src/mbgl/util/thread.hpp

```diff
@@ -87,6 +87,34 @@ class Thread {
     RunLoop* loop = nullptr;
 };
 
+inline std::string getCurrentThreadName() {
+#if defined(__APPLE__)
+    char name[32];
+    pthread_getname_np(pthread_self(), name, sizeof(name));
+    return name;
+#elif defined(__GLIBC__) && defined(__GLIBC_PREREQ)
+#if __GLIBC_PREREQ(2, 12)
```

Comment:
Shouldn't this be part of the `elif` condition on the previous line? Otherwise the function will be missing a `return` statement in the case that you are compiling with a glibc that doesn't meet the prerequisite.